### PR TITLE
fixes #100 - pasting at EOF should create new line

### LINKED ISSUE #100: putting linewise text into a register is broken at EOF
For example, in a file with just the text `abc` (no newline), `yyp` should result in the file:

```
abc
abc
```

Since there's no newline in the original file, there has to be a special case to add the newline into the register with the yanked text.

### PR DESCRIPTION
**Pull request by guillermooo**
I think ViPasteRight needs to account for the case where the sel is at EOF to position the caret at the beginning of the last line. At the moment it incorrectly places it after EOL of the last line that existed before the paste operation.

### CODE REVIEW

**Comment by guillermooo** [comment]
Thanks for the feedback. I'm not sure how to check whether we're in visual line mode/visual characterwise selection and at the same time make `yy` still insert a new line when it's the last line of the buffer.

**Comment by guillermooo** [comment]
Btw, if #104 is accepted, I will have to redo this against those changes.

File: vintage.py [hunk 1]
```diff
@@ -902,8 +916,14 @@ def run(self, edit, register, repeat = 1, forward = True):
...
 
-            if len(text) > 0 and text[-1] == '\n':
+            if text.endswith('\n'):
                 # paste line-wise
+                # If we are at the end of the buffer, add a new line as
+                # Vim does.
+                if (self.view.rowcol(s.end())[0] == \
+                    self.view.rowcol(self.view.size())[0]):
+                        text = '\n' + text[:-1]
+
```
Comment (misfo) [inline vintage.py:926]: This last set of changes to pasting looks good. Can you submit this as a separate fix (for #117)?

**Comment by misfo** [comment]
I'm thinking the fix for #100 can be a lot simpler (the fix for #117 is great!) If the motion mode indicates a linewise motion/selection, ensure what's being put in the register has a trailing newline. So we don't really have to have special EOF conditions; we'll just always check that linewise yanks always have the newline.

**Comment by guillermooo** [comment]
Deprecating this patch. There will be a new series of patches addressing operations at EOF and other newline-related problems when yanking/deleting/pasting.

**Pull request closed without merging**